### PR TITLE
feat: add SmartcarVehicleRequest class and Vehicle request method

### PR DESCRIPTION
**Pull request by naomiperez**
Add a new general purpose/BSE related object/method SmartcarVehicleRequest to allow making requests like so: `SmartcarVehicleRequest req = new SmarcarVehicleRequest.Builder.method("GET").path("tesla/thermometer/outside").build();` Adds request() method to Vehicle class. Adds E2E tests for this class and method.

Test Plan: create an app with my Smartcar Developer account and make requests with this method once it's in staging/prod.

Asana: https://app.asana.com/0/1201332815308984/1201631034519423/f

### CODE REVIEW

File: src/main/java/com/smartcar/sdk/Vehicle.java [hunk 1]
```diff
@@ -359,6 +361,59 @@ public BatchResponse batch(String[] paths) throws SmartcarException {
...
+   *
+   * @param vehicleRequest with options for this request. See Smartcar.SmartcarVehicleRequest
+   * @return the VehicleResponse object containing the response from the requested endpoint
+   * @throws SmartcarException if the request is unsuccessful
+   */
+  public VehicleResponse request(SmartcarVehicleRequest vehicleRequest) throws SmartcarException, IOException {
+    String version = this.version;
+    String vehicleId = this.vehicleId;
+
+    if(vehicleRequest.getVersion() != null) {
```
[the remark this answers is not in the document]
Comment (naomiperez) [reply src/main/java/com/smartcar/sdk/Vehicle.java:376]: I'll run the google-java-format linter for the new commits

[the remark this answers is not in the document]
Comment (naomiperez) [reply src/main/java/com/smartcar/sdk/Vehicle.java:376]: Ran into an issue with this that I'm trying to resolve - the minimum supported runtime version for `google-java-format` is JDK 11 (https://github.com/google/google-java-format/releases/tag/google-java-format-1.8). We use JDK 8, so I'm going to see if an old version of `google-java-format` would work, or if there are other options (like a different program or IntelliJ plugin).

[the remark this answers is not in the document]
Comment (naomiperez) [reply src/main/java/com/smartcar/sdk/Vehicle.java]: Update: found that older versions (1.7) of `google-java-format` works so I used that.

**Review by rsimari — APPROVED** [review]
Looks good to me! There are tons of white space changes here, probably due to a linter. Lets discuss whether or not we want these in this PR

**Pull request closed without merging**